### PR TITLE
feat(table): added column sort

### PR DESCRIPTION
**Pull request by bryancboyd**
**Summary**

- Added column sort (basic lexicographic compare, no custom sorts yet)

image

**Change List (commits, features, bugs, etc)**

- (see commits in PR)

**Related Issues**

- (none)

**Acceptance Test (how to verify the PR)**

- View the "Table --> sort" story in storybook, verify that the 1st, 3rd, and 4th columns are sortable.

### CODE REVIEW

File: src/components/Table/Table.js [hunk 1]
```diff
@@ -172,13 +184,24 @@ const Table = ({ columns, data, view, actions, options }) => {
...
+                    style={filterBarActive === true ? filterBarActiveStyle : {}}
+                    isSortable={column.isSortable}
+                    isSortHeader={hasSort}
+                    onClick={() => {
+                      if (column.isSortable && actions.table.onChangeSort) {
+                        actions.table.onChangeSort(column.id);
+                      }
+                    }}
+                    sortDirection={hasSort ? view.table.sort.direction : 'NONE'}>
+                    <span className="bx--table-header-label">{column.name}</span>
```
Comment (tay1orjones) [inline src/components/Table/Table.js]: We can just use `{column.name}` without the wrapper `span` here, [`<TableHeader/>` renders children in a span](https://github.com/IBM/carbon-components-react/blob/master/src/components/DataTable/TableHeader.js#L79)

Comment (bryancboyd) [reply src/components/Table/Table.js]: my bad, I missed that when merging.  I've pushed a change onto the PR.

**Pull request merged**